homotopy-model/src/proof.rs

```diff
@@ -990,25 +1077,42 @@ impl Default for RenderStyle {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize)]
 pub struct AttachOption {
     pub generator: Generator,
+    pub diagram: DiagramN,
+    pub tag: Option<&'static str>,
```

Comment:
This is the bit that is preventing us from deriving Deserialize for AttachOption. We should probably be able to change this to Option<String>, at the price of a few clones.

Reply:
Go for it.